### PR TITLE
Feature/ add server tests for tree

### PR DESCRIPTION
**Pull request by Mohammad96Assaf**
ref #169

### CODE REVIEW

File: internal/server/http/handler/v1/tree/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+
+		// then
+		assert.Nil(t, err)
+		assert.Equal(t, http.StatusInternalServerError, resp.StatusCode)
+		mockService.AssertExpectations(t)
+	})
+
+}
+
+func getMockTreeUpdateRequest() *httpEntities.TreeUpdateRequest {
```
Comment (doriengr) [inline internal/server/http/handler/v1/tree/handler_test.go:432]: getMockTreeUpdateRequest() and getMockTreeCreateRequest() can be unified in one.

**Review by doriengr — COMMENTED** [review]
The test for the `routes.go` file is missing.

**Comment by Mohammad96Assaf** [comment]
@doriengr please approve it

File: internal/server/http/handler/v1/tree/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+)
+
+func TestGetTreeByID(t *testing.T) {
+	t.Run("should return tree successfully", func(t *testing.T) {
+		app := fiber.New()
+		mockTreeService := serviceMock.NewMockTreeService(t)
+		app.Get("v1/tree/:id", GetTreeByID(mockTreeService))
+
+		treeID := int32(1)
+		mockTree := &entities.Tree{
```
Comment (doriengr) [inline internal/server/http/handler/v1/tree/handler_test.go:29]: Can this also be data from the utils?

Comment (doriengr) [reply internal/server/http/handler/v1/tree/handler_test.go:29]: I am going to approve the MR, but please check before merging, if you can use the test data here.

**Review by doriengr — APPROVED** [review]

**Pull request merged**